src/app/products/charges/create-charge/create-charge.component.ts

```diff
@@ -0,0 +1,215 @@
+          this.chargeTimeTypeData = this.chargesTemplateData.shareChargeTimeTypeOptions;
+          break;
+      }
+    });
+  }
+
+  /**
+   * @returns {any} Filtered charge calculation type data.
+   */
+  filteredChargeCalculationType(): any {
+    return this.chargeCalculationTypeData.filter((chargeCalculationType: any) => {
+      if (this.chargeForm.get('chargeTimeType').value === 12 && (chargeCalculationType.id === 3 || chargeCalculationType.id === 4)) {
+        return false;
+      }
+      if (this.chargeForm.get('chargeTimeType').value !== 12 && chargeCalculationType.id === 5) {
+        return false;
+      }
+      if (this.chargeForm.get('chargeAppliesTo').value === 2) {
+        if (!(this.chargeForm.get('chargeTimeType').value === 5 || this.chargeForm.get('chargeTimeType').value === 16) && chargeCalculationType.id === 2) {
+          return false;
+        }
+      }
+      return true;
+    });
+  }
+
+  /**
+   * Sets the conditional controls of the user form
+   */
+  setConditionalControls() {
+    this.chargeForm.get('chargeAppliesTo').valueChanges.subscribe((chargeAppliesTo) => {
+      switch (chargeAppliesTo) {
+        case 1: // Loan
+          this.chargeForm.addControl('chargePaymentMode', new FormControl('', Validators.required));
+          this.chargeForm.removeControl('incomeAccountId');
+          break;
+        case 2: // Savings
+          this.chargeForm.removeControl('chargePaymentMode');
+          this.chargeForm.removeControl('incomeAccountId');
+          break;
+        case 3: // Client
+          this.chargeForm.removeControl('chargePaymentMode');
+          this.chargeForm.addControl('incomeAccountId', new FormControl(''));
+          break;
+        case 4: // Shares
+          this.chargeForm.removeControl('chargePaymentMode');
+          this.chargeForm.removeControl('incomeAccountId');
+          this.chargeForm.get('penalty').setValue(false);
+          break;
+      }
+      this.chargeForm.get('chargeCalculationType').reset();
+      this.chargeForm.get('chargeTimeType').reset();
+    });
+    this.chargeForm.get('chargeTimeType').valueChanges.subscribe((chargeTimeType) => {
+      this.chargeForm.removeControl('feeFrequency');
+      this.chargeForm.removeControl('feeInterval');
+      this.chargeForm.removeControl('feeOnMonthDay');
+      this.chargeForm.removeControl('addFeeFrequency');
+      switch (chargeTimeType) {
+        case 6: // Annual Fee
+          this.chargeForm.addControl('feeOnMonthDay', new FormControl('', Validators.required));
+          break;
+        case 7: // Monthly Fee
+          this.chargeForm.addControl('feeOnMonthDay', new FormControl(''));
+          this.chargeForm.addControl('feeInterval', new FormControl('', [Validators.required, Validators.min(1), Validators.max(12), Validators.pattern('^\\d*$')]));
+          this.repeatEveryLabel = 'Months';
+          break;
+        case 9: // Overdue Fee
+          this.chargeForm.get('penalty').setValue(true);
```

Comment:
Lock penalty in as true. (Disabled will also be applied on the HTML)

Reply:
Apply disabled from here only as I specified about the warning above.